google/resource_cloudfunctions_function.go

```diff
@@ -0,0 +1,462 @@
+				Default:  FUNCTION_DEFAULT_MEMORY,
+			},
+
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"timeout": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      DEFAULT_FUNCTION_TIMEOUT_IN_SEC,
+				ValidateFunc: validation.IntBetween(FUNCTION_TIMEOUT_MIN, FUNCTION_TIMEOUT_MAX),
+			},
+
+			"labels": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+
+			"storage_bucket": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"storage_object": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"trigger_bucket": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				Elem:          &schema.Schema{Type: schema.TypeString},
+				ConflictsWith: []string{"trigger_http", "trigger_topic"},
+			},
+
+			"trigger_http": {
+				Type:          schema.TypeBool,
+				Optional:      true,
+				ForceNew:      true,
+				Elem:          &schema.Schema{Type: schema.TypeBool},
+				ConflictsWith: []string{"trigger_bucket", "trigger_topic"},
+			},
+
+			"trigger_topic": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				Elem:          &schema.Schema{Type: schema.TypeString},
+				ConflictsWith: []string{"trigger_http", "trigger_bucket"},
+			},
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCloudFunctionsCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	service := config.clientCloudFunctions
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	region, err := getRegion(d, config)
+	if err != nil {
+		return err
+	}
+
+	if FUNCTION_ALLOWED_REGION[region] != true {
+		return fmt.Errorf("Invalid region. Now allowed only us-central1. See https://cloud.google.com/about/locations/")
+	}
+
+	funcName := d.Get("name").(string)
+
+	function := &cloudfunctions.CloudFunction{
+		Name: createCloudFunctionsPathString(CLOUDFUNCTIONS_FULL_NAME, project, region, funcName),
+	}
+
+	storageBucket := d.Get("storage_bucket").(string)
+	storageObj := d.Get("storage_object").(string)
+	function.SourceArchiveUrl = fmt.Sprintf("gs://%v/%v", storageBucket, storageObj)
+
+	if v, ok := d.GetOk("memory"); ok {
+		memory := v.(int)
+		if FUNCTION_ALLOWED_MEMORY[memory] != true {
+			return fmt.Errorf("Allowed values for memory are: 128MB, 256MB, 512MB, 1024MB, and 2048MB. "+
+				"Got %d", memory)
+		}
+		function.AvailableMemoryMb = int64(memory)
```

Comment:
Any reason to not keep the field in the schema with the same name as is in the API (AvailableMemoryMb vs memory)?

Reply:
Renamed